onnxruntime/core/session/provider_registration.cc

```diff
@@ -74,7 +75,15 @@ ORT_API_STATUS_IMPL(OrtApis::SessionOptionsAppendExecutionProvider,
 #endif
   } else if (strcmp(provider_name, "XNNPACK") == 0) {
 #if defined(USE_XNNPACK)
+    // If Xnnpack is enabled and user don't give a specific value, we need to set it same as cpu threadpool size
```

Comment:
Need to update the documentation in onnxruntime_c_api.h that has info on the provider options for EPs that can be enabled via SessionOptionsAppendExecutionProvider.

Also need to add something equivalent to https://onnxruntime.ai/docs/execution-providers/SNPE-ExecutionProvider.html for xnnpack on the gh-pages branch now that the XNNPACK EP has a decent number of kernels and it's meaningful for users to try it out.

Reply:
Will create another PR to update documents